src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -33,6 +33,7 @@
 import com.openshift.restclient.http.IHttpConstants;
 import com.openshift.restclient.utils.SSLUtils;
 
+import okhttp3.Authenticator;
```

Comment:
any progress for the abstraction ? 

PS: It's not a big deal to change the authenticator, when okhttp lib will really change ;)